### PR TITLE
Update to python-0.9.0 - only unix - with ARM support

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
```
/Users/runner/miniforge3/conda-bld/lancedb_1720331978333/test_tmp/conda_test_runner.sh: line 3: /Users/runner/miniforge3/conda-bld/lancedb_1720331978333/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/python: Bad CPU type in executable
```

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
```
We stopped hearing from agent Azure Pipelines 133. Verify the agent machine is running and has a healthy network connection. Anything that terminates an agent process, starves it for CPU, or blocks its network access can cause this error. For more information, see: https://go.microsoft.com/fwlink/?linkid=846610
```

**Comment by jan-janssen** [comment]
@conda-forge/core The Linux ARM tests timeout - most likely because they require too much memory. I previously had a similar issue on Windows https://github.com/conda-forge/lancedb-feedstock/pull/11 . Is there any way to get access to native ARM builds?

**Comment by h-vetinari** [comment]
Before we think about native arm runners, can you try adding
```yaml
azure:
  free_disk_space: true
  settings_linux:
    swapfile_size: 15GiB
```
to `conda-forge.yml` and rerender?

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
@h-vetinari Is there a similar trick for windows to fix the out of memory issue in https://github.com/conda-forge/lancedb-feedstock/pull/11 ?

**Comment by h-vetinari** [comment]
> @h-vetinari Is there a similar trick for windows to fix the out of memory issue in #11 ?

We don't have a common interface unfortunately, and windows works... less reliably... but in principle you can do:
```
azure:
  settings_win:
    variables:
      CONDA_BLD_PATH: C:\\bld\\  # <- potentially not necessary, but C: tends to have more space (~70 GB) available
      SET_PAGEFILE: "True"
```
This will set a swapfile somewhere [between](https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/320) 8GB & 16GB, depending what the system returns.

Here's some other related issues: https://github.com/conda-forge/conda-smithy/issues/1949 & https://github.com/conda-forge/conda-smithy/issues/1971

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,49 +1,57 @@
...
+    - maturin >=1.4
     - pip
   run:
-    - python >=3.8
+    - python
     - deprecation
-    - pylance ==0.9.18
-    - ratelimiter >=1.0,<2.dev0
+    - pylance ==0.13.0
+    # # ratelimiter package doesn't work on 3.11
```
Comment (jan-janssen) [inline recipe/meta.yaml:38]:
```suggestion
    # ratelimiter package doesn't work on 3.11
```

**Pull request closed without merging**